File: app/src/main/java/com/sample/dhruvmj/waveapp/model/Product.java
package com.sample.dhruvmj.waveapp.model;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import com.google.gson.annotations.SerializedName;

import java.text.NumberFormat;
import java.util.Locale;

/**
 * @author Dhruv
 */

public class Product extends BaseObservable {
    @SerializedName("name")
    private String productName;
    private int price;

    public Product(String productName, int price) {
        this.productName = productName;
        this.price = price;
    }

    @Bindable
    public String getProductName() {
        return productName;
    }

    @Bindable
    public String getPrice() {
        NumberFormat currencyFormat = NumberFormat.getCurrencyInstance(Locale.getDefault());
        return currencyFormat.format(price);
    }
}

File: app/src/main/java/com/karlptrck/wavechallenge/model/Product.kt
package com.karlptrck.wavechallenge.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(tableName = "products")
data class Product(
    @PrimaryKey @field:SerializedName("id") val id : Int,
    @field:SerializedName("name") val name : String,
    @field:SerializedName("price") val price : Double
)